### PR TITLE
Puppet strings 4 all #757

### PR DESCRIPTION
**Pull request by alvagante**

### CODE REVIEW

File: lib/puppet/type/sensu_api_config.rb [hunk 1]
```diff
@@ -1,3 +1,14 @@
+# @!puppet.type.param name
```
Comment (ghoneycutt) [inline lib/puppet/type/sensu_api_config.rb]: you doing the rest of the types/providers in another pass?

Comment (alvagante) [inline lib/puppet/type/sensu_api_config.rb]: @ghoneycutt yes, but actually I was wondering if all the current types, and relevant providers, are going to remain as is or some of them are will be replaced by the works @jeffmccune is doing in https://github.com/sensu/sensu-puppet/pull/785 will affect some of the current ones, and their params. If so I wonder if it makes sense to prepare docs for types that are going to change soon.

Comment (ghoneycutt) [inline lib/puppet/type/sensu_api_config.rb]: Makes sense. Don't bother with the ones that are changing and we'll make another pass after the dust settles.

**Comment by alvagante** [comment]
@ghoneycutt This is ready for merging, from my point of view. I've added Puppet 4 params validations in a pair of classes where it was missing, in order to have valid entries in puppet strings and to align those classes to the others.

**Comment by ghoneycutt** [comment]
Thanks @alvagante 

Released in v2.31.0

**Pull request merged**